sim/lib/mobilitysim.py

```diff
@@ -621,15 +677,21 @@ def simulate(self, max_time, seed=None):
         if self.verbose:
             print(f'Simulated {len(all_mob_traces)} visits.', flush=True)
 
-        # Find the contacts in all sites in the histories
-        if self.verbose:
-            print(f'Find contacts... ', end='')
-        self.contacts = self._find_contacts(all_mob_traces)
-        # FIXME: contact_count calculation takes too long
-        # self.contact_count = sum(len(self.contacts[i][j]) for i in range(
-        #     self.num_people) for j in range(self.num_people))
-        # if self.verbose:
-        #     print(f'Found {self.contact_count} contacts', flush=True)
+        if not dynamic_tracing:
+            # Find the contacts in all sites in the histories
+            if self.verbose:
+                print(f'Find contacts... ', end='')
+            self.contacts = self._find_contacts(all_mob_traces)
+            # FIXME: contact_count calculation takes too long
```

Comment:
`all_mob_traces` needs to be stores in this part of the if clause as well to be consistent